src/browserWindow.ts

```diff
@@ -0,0 +1,144 @@
+import * as electron from 'electron'
+import { VibrancyOptions } from './vibrancy';
+import debug from './debug';
+import { isRS4OrGreater, isWindows10 } from './os';
+import { getConfigFromOptions, rgbToHex, Vibrancy, VibrancyConfig, _setVibrancy } from './vibrancy';
+
+/**
+ * Allow modifying default BrowserWindowConstructorOptions
+ * to change vibrancy to VibrancyOptions.
+ */
+type Modify<T, R> = Omit<T, keyof R> & R;
+
+type _AcrylicBrowserWindowConstructorOptions = Modify<electron.BrowserWindowConstructorOptions, {
+	/**
+	 * The vibrancy settings for the window. Can be
+	 * a VibrancyTheme or the VibrancyOptions object.
+	 */
+	vibrancy?: Vibrancy
+}>;
+
+/**
+ * The options of the BrowserWindow constructor including the vibrancy options.
+ */
+// The intelisense of type _AcrylicBrowserWindowConstructorOptions is shown as the raw type, instead of the actual type name.
+// tslint:disable-next-line
+export interface AcrylicBrowserWindowConstructorOptions extends _AcrylicBrowserWindowConstructorOptions { }
+
+export interface WindowConfig {
+	vibrancyActivated: boolean
+	vibrnacyConfig: VibrancyConfig
+	targetOpacity: number
+	opacity: number
+	opacityInterval: any
+	currentOpacity: number
+	moveTimeout: any
+}
+
+/**
+ * Class wrapper for `electron.BrowserWindow`.
+ * The class handles the vibrancy effects.
+ */
+export class BrowserWindow extends electron.BrowserWindow {
+	/**
+	 * Set the vibrancy for the specified window.
+	 *
+	 * @param options
+	 */
+	// https://github.com/microsoft/TypeScript/issues/30071
+	// @ts-ignore
+	setVibrancy(options?: Vibrancy) {
+		if (options) {
+			this.#winconfig.vibrnacyConfig = getConfigFromOptions(options);
+			_setVibrancy(this, this.#winconfig.vibrnacyConfig);
+		} else {
+			// If disabling vibrancy, turn off then save
+			_setVibrancy(this)
+			this.#winconfig.vibrnacyConfig = getConfigFromOptions(undefined);
+		}
+	}
+
+	#vibconfig = getConfigFromOptions(this.options?.vibrancy)
+
+	#winconfig: WindowConfig = {
+		targetOpacity: 0,
+		vibrancyActivated: false,
+		vibrnacyConfig: this.#vibconfig,
+		opacity: 0,
+		currentOpacity: 0,
+		opacityInterval: undefined,
+		moveTimeout: undefined
+	}
```

Comment:
> Please provide a detailed description of the following properties in options.
> 
> * [ ]  currentOpacity
> * [ ]  opacity
> * [ ]  opacityInterval
> * [ ]  targetOpacity

Reply:
opacity is user-set opacity.

All of the remaining variables are designed to control the transparency of focus.
CurrentOpacity is the transparency value that Windows actually has currently.
TargetOpacity is the opacity that Windows should have.
opacityInterval is the id of the interval for smooth transition of currentOpacity to targetOpacity when focus changes. The reason for saving this is to call clearInterval after the conversion is complete.